app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,202 @@
+
+  height += parseInt(style.marginTop) + parseInt(style.marginBottom)
+  return height
+}
+
+class CheckboxSearchFilter {
+  constructor (params) {
+    this.params = params
+    this.container = params.container
+    this.container.classList.add('app-checkbox-filter--enhanced')
+    this.checkboxes = this.container.querySelectorAll("input[type='checkbox']")
+    this.checkboxesContainer = this.container.querySelector('.app-checkbox-filter__container')
+    this.checkboxesInnerContainer = this.checkboxesContainer.querySelector('.app-checkbox-filter__container-inner')
+    this.setup()
+  }
+
+  filterCheckboxes () {
+    const textValue = this.sanitizeText(this.textBox.value)
+    const allCheckboxes = this.getAllCheckboxes()
+
+    for (let i = 0; i < allCheckboxes.length; i++) {
+      const labelValue = this.sanitizeText(allCheckboxes[i].querySelector('.govuk-checkboxes__label').textContent)
+      if (labelValue.search(textValue) === -1) {
+        allCheckboxes[i].style.display = 'none'
+      } else {
+        allCheckboxes[i].style.display = 'block'
+      }
+    }
+
+    this.updateStatusBox({
+      foundCount: this.getAllVisibleCheckboxes().length,
+      checkedCount: this.getAllVisibleCheckedCheckboxes().length
+    })
+  }
+
+  getAllCheckboxes () {
+    return this.checkboxesContainer.querySelectorAll('.govuk-checkboxes__item')
+  }
+
+  getAllVisibleCheckboxes () {
+    return [].filter.call(this.getAllCheckboxes(), function (el) {
+      return el.style.display === 'block'
+    })
+  }
+
+  getAllVisibleCheckedCheckboxes () {
+    return [].filter.call(this.getAllVisibleCheckboxes(), function (el) {
+      return el.querySelector('.govuk-checkboxes__input').checked
+    })
+  }
+
+  getTextBoxHtml () {
+    const containerId = this.container.id
+    const label = document.createElement('label')
+    label.setAttribute('for', containerId + '-checkbox-filter__filter-input')
+    label.className = 'govuk-label govuk-visually-hidden'
+    label.innerHTML = this.params.textBox.label
+
+    const input = document.createElement('input')
+    input.className = 'app-checkbox-filter__filter-input govuk-input'
+    input.setAttribute('id', containerId + '-checkbox-filter__filter-input')
+    input.setAttribute('type', 'text')
+    input.setAttribute('aria-describedby', containerId + '-checkboxes-status')
+    input.setAttribute('aria-controls', containerId + '-checkboxes')
+    input.setAttribute('autocomplete', 'off')
+    input.setAttribute('spellcheck', 'false')
+
+    return [label, input]
+  }
+
+  getVisibleCheckboxes () {
+    const visibleCheckboxes = [].filter.call(this.checkboxes, (el) => this.isCheckboxInView(el))
+
+    // add an extra checkbox, if the label of the first is too long it collapses onto itself
+    visibleCheckboxes.push(this.checkboxes[visibleCheckboxes.length - 1])
+    return visibleCheckboxes
+  }
+
+  isCheckboxInView (checkbox) {
+    const initialOptionContainerHeight = this.checkboxesContainer.style.height
```

Comment:
Have you checked this part of the functionality works in a few older browsers (both desktop or device)?

Reply:
@benswannack this is the main thing I'm not sure about as I can't find a browser version specific compatibility list.

Reply:
I've tested on all the browsers from this list and it looks good! https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in-from-january-2021